### PR TITLE
Fix bugs and warnings caught by lgtm

### PR DESCRIPTION
**Pull request by idshibanov**
It caught a critical typo in `settings.cpp` and `text.cpp`. `Troops` class allocates memory so it needs an explicit assignment operator.

Also fixing the straightforward warnings around the project.
https://lgtm.com/projects/g/ihhub/fheroes2/alerts/?mode=list&severity=

### CODE REVIEW

File: src/fheroes2/maps/maps_tiles.cpp [hunk 1]
```diff
@@ -135,6 +135,14 @@ std::string Maps::TilesAddon::String( int lvl ) const
     return os.str();
 }
 
+Maps::TilesAddon::TilesAddon( const Maps::TilesAddon & ta )
+    : uniq( ta.uniq )
+    , level( ta.level )
+    , object( ta.object )
+    , index( ta.index )
+    , tmp( 0 )
```
Comment (idshibanov) [inline src/fheroes2/maps/maps_tiles.cpp:143]: Judging by the use of this tmp variable I decided that it should stay consistent with the assignment operator and shouldn't copy it.

**Comment by idshibanov** [comment]
Static analysis complains about tinyxml changes since those files are unformatted.

@ihhub I can either format those files based on clang or we can ignore those. What do you think? Do we need them at all in the future?

**Comment by ihhub** [comment]
> Static analysis complains about tinyxml changes since those files are unformatted.
> 
> @ihhub I can either auto-format those files based on clang or we can ignore those. What do you think? Do we need them at all in the future?

Ignore hem please, they are third party library files and also we don't use xml in our code. It is a legacy code.

**Comment by ihhub** [comment]
@idshibanov good job as usual!

**Comment by ihhub** [comment]
By the way, **lgtm** takes up to 24 hours to update status so we need to wait some time to get fresh report.

**Comment by idshibanov** [comment]
> By the way, **lgtm** takes up to 24 hours to update status so we need to wait some time to get a fresh report.

Yes, I already guessed that's the case :) great tool anyway, I like the reports

**Comment by ihhub** [comment]
I ran [Coverity](https://scan.coverity.com/projects/ihhub-fheroes2?tab=overview) scan as well. Sadly only me can view the details. There are many issues found by the tool.

**Comment by ihhub** [comment]
@idshibanov , you could try to create an account in Coverity and send me an invitation to the project so you might be able to see the results if you are interested in fixing all these issues as well.

**Comment by idshibanov** [comment]
> @idshibanov , you could try to create an account in Coverity and send me an invitation to the project so you might be able to see the results if you are interested in fixing all these issues as well.

Signed up and got the project access. Thank you. Looks like new lgtm report is already in progress.

**Pull request merged**